File: java/amp-examples/stream-subscriber/src/main/java/amp/examples/streaming/subscriber/CollectionHandler.java
package amp.examples.streaming.subscriber;

import amp.examples.streaming.common.ModernMajorGeneralMessage;
import cmf.bus.Envelope;
import cmf.eventing.patterns.streaming.IStreamingCollectionHandler;
import cmf.eventing.patterns.streaming.StreamingEventItem;

import java.util.Collection;
import java.util.Map;

public class CollectionHandler implements IStreamingCollectionHandler<ModernMajorGeneralMessage> {

    @Override
    public void handleCollection(Collection<StreamingEventItem<ModernMajorGeneralMessage>> events) {
        System.out.println("Received a collection from AMPere of size: " + events.size());

        for(StreamingEventItem<ModernMajorGeneralMessage> eventItem : events) {
            System.out.println("Event Item: SequenceId => " + eventItem.getSequenceId() +
                    ", Position => " + eventItem.getPosition() +
                    "\n\t Content => " + eventItem.getEvent().getContent());
        }

        System.out.println("Processing complete.");
    }

    @Override
    public void onPercentCollectionReceived(double percent) {
        System.out.println("Percent of events received: " + percent + "%");
    }

    @Override
    public Class<ModernMajorGeneralMessage> getEventType() {
        return ModernMajorGeneralMessage.class;
    }

}

File: dotnet/examples/amp.examples.streaming.subscriber/CollectionHandler.cs
using amp.examples.streaming.common;
using cmf.eventing.patterns.streaming;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace amp.examples.streaming.subscriber
{
    public class CollectionHandler : IStreamingCollectionHandler<ModernMajorGeneralMessage>
    {
        

        public void HandleCollection(IEnumerable<StreamingEventItem<ModernMajorGeneralMessage>> events)
        {
            Console.WriteLine("Received a collection from AMPere of size: " + events.Count());

            foreach(StreamingEventItem<ModernMajorGeneralMessage> eventItem in events)
            {
                Console.WriteLine(string.Format("Event Item: SequenceId => {0}, Position => {1}, Content => {2}",
                    eventItem.SequenceId, eventItem.Position, eventItem.Event.Content));
            }

            Console.WriteLine("Processing complete.");
        }
       
        public Type EventType
        {
            get { return typeof(ModernMajorGeneralMessage); }
        }

        public void OnPercentCollectionReceived(double percent)
        {
            Console.WriteLine("Percent of events received: " + percent + "%");
        }
    }
}
